test/linalg.jl

```diff
@@ -53,6 +53,43 @@ using Test
             end
         end
 
+        @testset "eigen decomposition" begin
+            # this hits both eigen(BlockDiagonal) and blockEigen, since the former uses the latter
+            evalsBlock, evecsBlock = eigen(b1)
+            evals, evecs = eigen(Matrix(b1))
+            @static if VERSION < v"1.2"
```

Comment:
Is it possible to put this transformation in a separate function? Also, it is not clear immediately why this block is sorting `abs2` and the next one isn't?

Reply:
Here the eigenvalues are complex, because the Matrix is not Symmetric/Hermitian. This means we need to convert to real numbers somehow to establish an order. I can write this in a comment about the abs2 if you wish.
If I want to extract a function for the tests (which makes sense), where do I put it? Just inline in the `@test` block?

I'm generally not perfectly happy with these tests because they can break when there degenerate eigenvalues. For our matrices here, this does not happen (I checked). Also in general Random Matrix theory asserts that random matrices are very unlikely to have degenerate eigenvalues. However, if in the future someone decides to implement theses tests with degenerate eigenvalues. I'm not sure whether we should try to think of way to circumvent this issue. 
It's probably too much effort for too little reward.

Reply:
Thanks, makes sense! It would be great to have a comment to clarify this for future readers. I don't feel strongly about making it a separate function however